### PR TITLE
[Improvement-10760][dolphinscheduler-api] Some variables are redundant and fix checkstyle

### LINKED ISSUE #10760: [Feature][dolphinscheduler-api] Some variables are redundant
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Subclasses that inherit BaseServiceImpl and have the resourcePermissionCheckService attribute should remove this attribute in the subclass. Because this attribute already exists in the BaseServiceImpl  class  (eg:DataSourceServiceImpl ,ProjectServiceImpl)

### Use case

remove   the resourcePermissionCheckService attribute

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuchanghai**
## Brief change log
Some variables are redundant and fix checkstyle

## Verify this pull request

This pull request is code cleanup without any test coverage.

this close #10760

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 1]
```diff
@@ -455,60 +455,60 @@ public Map<String, Object> updateUser(User loginUser, int userId,
...
-//
-//            try {
-//                storageOperate.createTenantDirIfNotExists(newTenant.getTenantCode());
-//            } catch (Exception e) {
-//                logger.error("create tenant {} failed ,the reason is {}", newTenant, e.getMessage());
-//            }
-//            }
-//            user.setTenantId(tenantId);
-//        }
+        //        if (user.getTenantId() != tenantId) {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java]: please remove if not use.

Comment (fuchanghai) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java]: thx you advice

**Review by caishunfeng — APPROVED** [review]
LGTM if CI pass

**Comment by fuchanghai** [comment]
@SbloodyS

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**